OneCommunityGlobal/HGNRest
# Shengwei create setup history modal

# Description
<img width="622" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/70614111-ebda-4540-9863-3170c0186741">


## Related PRS (if any):
This backend PR is related to the frontend [PR1992.](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1992)

## Main changes explained:
- Add fetch and update (refresh, cancel) setup token API in profileInitialSetupController.js
- Update weekly task: deleteExpiredTokens to allow 90-day data retention for expired or cancelled tokens.

## How to test:
1. check into current branch and the frontend [PR1992.](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1992)
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Setup backend Gmail Oauth and .env in the backend: [Link for video tutorial ](https://docs.google.com/document/d/1HJJTDtWykjGb4IfiIFlHXDfp_IABvmwKxsMPqqF6rG8/edit) and update local env file
5. Refer to frontend [PR1992.](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1992) for testing

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/6c86eb1e-8d0b-42db-9c15-28b18c81366b



## Note:
You need to Setup gmail oauth feature for testing. Also, change the email variable in the **refreshSetupInvitation** function in **src/controllers/profileInitialSetupController.js** (backend). Comment out `const { email } = result;` and use `const email = 'YOUR_EMAIL_TO_RECEIVE_MSG@gmail.com';` so that the application will send email to the email you defined!
